csv-extractor/csv_extractor.py

```diff
@@ -155,25 +152,46 @@ def process_csv_file(client, monitor, csv_path, existing_time_series):
                 target=_log_error, args=(client.datapoints.post_multi_time_series_datapoints, current_time_series)
             )
         )
-    end_time_1 = time.time()
-    logger.debug("Time taken to process file " + str(csv_path) + " " + str((end_time_1 - start_time)))
-    [t.start() for t in network_threads]
-    [t.join() for t in network_threads]
-    logger.debug("Time take to complete network requests & total time to ingest file  " + str(csv_path) + " " +
-                str(time.time()-end_time_1) + " " +
-                 str(time.time() - start_time))
 
-    return count_of_data_points, len(unique_external_ids)
+    return count_of_data_points, len(unique_external_ids), network_threads, csv_path
+
+
+def post_all_data(q, monitor, remaining_files_count):
+    all_threads = []
+    total_count_data_points = 0
+    total_count_ext_ids = 0
+    all_paths = []
+    for item in list(q.queue):
+        count_data_points, count_ext_ids, network_threads, csv_path = item
+        all_threads+=network_threads
+        total_count_data_points+=count_data_points
+        total_count_ext_ids+=count_ext_ids
+        all_paths.append(csv_path)
+
+    [t.start() for t in all_threads]
+    [t.join() for t in all_threads]
+    for p in all_paths:
+        remaining_files_count -= 1
+        monitor.unprocessed_files_gauge.set(remaining_files_count)
+        monitor.successfully_processed_files_gauge.inc()
+
+    monitor.count_of_time_series_gauge.set(total_count_ext_ids)
+    monitor.incr_total_data_points_counter(total_count_data_points)
+    monitor.push()
+    [csv_path.unlink() for csv_path in all_paths]
 
 
 def process_files(client, monitor, paths, time_series_cache, failed_path) -> None:
     """Process one csv file at a time, and either delete it or possibly move it when done."""
     remaining_files_count = len(paths)
     monitor.successfully_processed_files_gauge.set(0)
+    queue = Queue(maxsize=20)
+    st = time.time()
     for path in paths:
         try:
             try:
-                count_data_points, count_ext_ids = process_csv_file(client, monitor, path, time_series_cache)
+                count_data_points, count_ext_ids, network_threads, csv_path  = process_csv_file(client, monitor, path, time_series_cache)
+                queue.put((count_data_points, count_ext_ids, network_threads, csv_path))
```

Comment:
path and csv_path holds the same value, no need to return it from process_csv_file.
you can also put it straight onto the queue.
queue.put(process_csv_file(client, monitor, path, time_series_cache) + (path, ))